newIDE/app/src/MainFrame/EditorContainers/DebuggerEditorContainer.js

```diff
@@ -66,6 +66,13 @@ export class DebuggerEditorContainer extends React.Component<
     }
   }
 
+  setDebuggerToolbar = node => {
+    if (!this.props.isActive) {
+      return;
+    }
+    this.props.setToolbar(node);
```

Comment:
if we use the `setToolbar` function, what is the `updateToolbar` for?

Reply:
This function `updateToolbar` is required on the interface of a XxxEditorContainer (i.e: the components being the tabs in `MainFrame`). This imperative method is called when a tab is focused:

See in MainFrame:
![image](https://user-images.githubusercontent.com/1280130/144400943-e0f94177-e71c-44e5-8694-ff92a23991d6.png)
![image](https://user-images.githubusercontent.com/1280130/144400993-48b695d2-ecda-419d-8db4-f47686ab9624.png)

In theory, `updateToolbar` is only called when the tab is focused, so it's safe to ask the children to update the toolbar (See `if (this.editor) this.editor.updateToolbar();` in this file).

Where when the children trigger an update to the toolbar, it will usually not cause any issue because this usually happen after a user interaction... but here the debugger can react to an outside signal. And so it would call setToolbar even if it's not active! 
The MainFrame will not protect against this and will happily replace the toolbar with the last one that called setToolbar.

So I wonder if the real fix is not to be done in MainFrame actually:
* Search for `setToolbar: setEditorToolbar`
* We should replace this by a call to `setEditorToolbar` ONLY if the tab is active. Otherwise we ignore it.
* Other calls to setEditorToolbar looks safe.